docs/README.md

```diff
@@ -0,0 +1,241 @@
+primers. If it is not you will have an error. You must have 2 blank lines between a heading
+and whatever documentation/text is above it except when using the directive to create a 
+reference. The there need to be a single blank line above the heading and 2 blank lines above 
+the directive.
+
+
+### Code blocks
+
+--------------------------------------------------------------
+
+* No tabs are to be used in a code block
+* Indents are done using 4 spaces and only 4 spaces
+* 2 empty lines at the end of a clode block
+* One empty line between the code block directive and the code
+* `.. code-block:` is the only directive that should be used. `::`, `:code:` or `.. code:` should not be used
+* specify the language after the directive. Sopme examples are: `.. code-block: python`, `.. code-block: c`, `.. code-block: shell`, `.. code-block: make`
+* If you want to separate code into easier to understand sections you can do so with a single empty line. No more than ONE line.
+
+
+### bulleted lists
+
+-------------------------------------------------------------
+
+to create a bulleted list you do the follow
+
+
+    - item1: description
+    - item2: If you want to span multiple
+      lines it must be done like this
+    - item3: If you want to use a code block it must be done like this
+      
+      .. code-block: python
+
+          # this is some code
+
+    - item3: If you want to have several layers of bullets it needs to be done like this
+
+      - level 2 item 1: text
+      - level 2 item 2: text
+
+
+you must end all lists with 2 empty lines except when it is a nested list. Then you use a 
+single empty line. The same thing holds true for code blocks as well. if it is nested into 
+a list then a single empty line after. If the nested list or code block is at the end of the 
+first level then you need to use 2 empty lines
+
+
+### referencing portions of the API
+
+------------------------
+
+If you want to reference portions of the LVGL code from the documentation
+there are special primers/directives to do this...
+
+    :cpp:func:`lv_init`
+    :c:macro:`LV_USE_FLEX`
+    :cpp:type:`lv_event_t`
+    :cpp:enum:`_lv_event_t`
+    :cpp:enumerator:`LV_EVENT_ALL`
+    :cpp:struct:`lv_image_dsc_t`
+    :cpp:union:`lv_style_value_t`
+
+
+There is a special directive when wanting to use a more complex exspression. 
+For example when showing the arguments passed to a function
+
+    :cpp:expr:`lv_obj_set_layout(obj, LV_LAYOUT_FLEX)`
+
+you CANNOT have expressions that are like this...
+
+    :cpp:expr:`lv_obj_set_layout(obj, LV_LAYOUT_FLEX/GRID)`
+    :cpp:expr:`lv_obj_set_layout(obj, LV_LAYOUT_*)`
+    :cpp:expr:`lv_obj_set_layout(*obj, LV_LAYOUT_FLEX)`
+    :cpp:expr:`lv_obj_set_layout((lv_obj_t *)obj, LV_LAYOUT_FLEX)`
+    :cpp:expr:`lv_obj_set_layout(&obj, LV_LAYOUT_FLEX);`
+    :cpp:expr:`lv_obj_set_layout(obj, ...)`
+
+Those eare all invalid.
+
+I will be adding onto this readme, so keep an eye open for new information...
```

Comment:
Please shortly describe the internal build process, and highlight the custom non trivial parts.

Reply:
You want me to write something up for the internal build process of the documentation? Do you have any idea of how much that is? It would be a literal book, as in several hundred pages long. Doing a summation of what it does is only going to leave a person wondering what in the hell I am trying to convey. Best to not explain the actual process and only give the information that needs to be given and nothing more. There is no need to explain how the build system works. All they need to know is what is allowed and what is not allowed. I don't believe I am able to highlight sections of text that is inside of a code block. I know I can do that with sphinx but I do not believe that is something that is able to be done using markdown.

Reply:
Don't need to go into the very detail. I mean e.g. We use Doxygen, breath converts it to Sphinx, where the relevant scripts are, link and a few words about the CI, etc. 

The goal is to move this information out of your head to the public a little 😄

Reply:
NO NO NO. If I do that then I have no job security... LOL